Card.h

```diff
@@ -0,0 +1,55 @@
+#ifndef CARD_H
+#define CARD_H
+
+class Card
+{
+public:
+    //two enums to define the card suits and ranks
+    enum Suit
+    {
+        clubs,
+        diamonds,
+        hearts,
+        spades,
+
+        max_suits,
+    };
+
+    enum Rank
+    {
+        rank_2,
+        rank_3,
+        rank_4,
+        rank_5,
+        rank_6,
+        rank_7,
+        rank_8,
+        rank_9,
+        rank_10,
+        rank_jack,
+        rank_queen,
+        rank_king,
+        rank_ace,
+
+         max_rank,
+    };
+
+private:
+    Suit m_suit{};
+    Rank m_rank{};
+
+public:
+    //deftault constructor for a card
+    Card() = default;
```

Comment:
You probably don't even want a default constructor at all. If I understand enums correctly, this will always give you a 2 of clubs which isn't particularly useful.